### PR TITLE
executor: fix bug about topn pushing down.

### PR DESCRIPTION
**Pull request by hanfei1991**
make right pb for topn and index double read.
@shenli @coocood @zimulala PTAL

### CODE REVIEW

File: executor/distsql.go [hunk 1]
```diff
@@ -585,13 +585,15 @@ func (e *XSelectIndexExec) doIndexRequest() (distsql.SelectResult, error) {
...
 	selIdxReq.IndexInfo = distsql.IndexToProto(e.table.Meta(), e.indexPlan.Index)
-	if len(e.indexPlan.SortItemsPB) > 0 {
-		selIdxReq.OrderBy = e.indexPlan.SortItemsPB
-	} else if e.indexPlan.Desc {
+	if e.indexPlan.Desc {
 		selIdxReq.OrderBy = []*tipb.ByItem{{Desc: e.indexPlan.Desc}}
 	}
-	if e.singleReadMode || e.where == nil {
-		// TODO: when where condition is all index columns limit can be pushed too.
+	if e.singleReadMode {
```
Comment (coocood) [inline executor/distsql.go]: Add more comment about the change.

File: executor/distsql.go [hunk 2]
```diff
@@ -729,6 +731,9 @@ func (e *XSelectIndexExec) doTableRequest(handles []int64) (distsql.SelectResult
 	selTableReq := new(tipb.SelectRequest)
 	if e.indexPlan.OutOfOrder {
 		selTableReq.Limit = e.indexPlan.LimitCount
+		if len(e.indexPlan.SortItemsPB) > 0 {
```
Comment (coocood) [inline executor/distsql.go]: This check can be removed?

File: executor/distsql.go [hunk 3]
```diff
@@ -585,13 +585,15 @@ func (e *XSelectIndexExec) doIndexRequest() (distsql.SelectResult, error) {
...
-		selIdxReq.OrderBy = e.indexPlan.SortItemsPB
-	} else if e.indexPlan.Desc {
+	if e.indexPlan.Desc {
 		selIdxReq.OrderBy = []*tipb.ByItem{{Desc: e.indexPlan.Desc}}
 	}
-	if e.singleReadMode || e.where == nil {
-		// TODO: when where condition is all index columns limit can be pushed too.
+	if e.singleReadMode {
+		selIdxReq.Limit = e.indexPlan.LimitCount
+		if len(e.indexPlan.SortItemsPB) > 0 {
```
Comment (coocood) [inline executor/distsql.go]: add comment about prevent overwrite `Desc`

File: executor/distsql.go (hunk 2 above)
Comment (hanfei1991) [inline executor/distsql.go]: no, if len(***) = 0, desc may be true.

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**